source/unix/opensslcrypto_hmac.c

```diff
@@ -47,38 +81,19 @@ struct aws_hmac *aws_sha256_hmac_default_new(struct aws_allocator *allocator, co
         return NULL;
     }
 
-#if OPENSSL_VERSION_LESS_1_1
-    HMAC_CTX_init(ctx);
-#endif
+    if (g_aws_openssl_hmac_ctx_1_0_table != NULL) {
```

Comment:
Could we just use [unsigned int OPENSSL_version_minor()](https://www.openssl.org/docs/manmaster/man3/OPENSSL_version_major.html) and avoid all this dlsym rigamarole?

Reply:
That wouldn't help with the conditional compilation problem since the public contract/functions change.

Reply:
If we're supporting both openssl 1.0 and 1.1, can we just call the deprecated 1.0 functions, and not call any of the 1.1-exclusive functions? when we stop supporting openssl 1.0 we can remove the deprecated calls and just call the new 1.1 functions.

I'm not familiar with what any of these functions do. I'm just hoping that the openssl authors are respecting the major version and not breaking 1.0-style code

Reply:
They're not deprecated, they're GONE, unfortunately.

Reply:
it's a fully breaking API change, requires conditional handling.